src/main/java/com/pinterest/secor/parser/ThriftMessageParser.java

```diff
@@ -52,10 +56,21 @@ public String getFieldName() {
         }
     }
 
-    public ThriftMessageParser(SecorConfig config) {
+    public ThriftMessageParser(SecorConfig config)
+            throws InstantiationException, IllegalAccessException,
+            ClassNotFoundException {
         super(config);
-        mDeserializer = new TDeserializer();
-        mThriftPath = new ThriftPath(mConfig.getMessageTimestampName(), (short) mConfig.getMessageTimestampId());
+        TProtocolFactory protocolFactory = null;
+        String protocolName = mConfig.getThriftProtocolClass();
+        
+        if (StringUtils.isNotEmpty(protocolName)) {
+            String factoryClassName = protocolName.concat("$Factory");
+            protocolFactory = ((Class<? extends TProtocolFactory>) Class.forName(factoryClassName)).newInstance();
+        } else
+            protocolFactory = new TBinaryProtocol.Factory();
```

Comment:
We are still using thrift 0.5, the code is different in ver 0.5.  I think we should use the constructor with no argument and let the underlying  thrift library handle the version difference.